core/src/main/java/com/linecorp/armeria/common/ConcurrentAttributes.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common;
+
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.annotation.UnstableApi;
+
+import io.netty.util.AttributeKey;
+
+/**
+ * An {@link Attributes} supporting concurrency of retrievals and updates.
+ */
+@UnstableApi
+public interface ConcurrentAttributes extends AttributesGetters, AttributesSetters {
```

Comment:
`Concurrent` doesn't imply mutability. How about `MutableAttributes`?

Reply:
I didn't think `MutableAttributes` implies supporting concurrent modifications.
For example `MutableList` in Scala and Kotlin are not thread-safe.
Immutable data structures are thread-safe by default so, I thought, users could infer mutability from `Concurrent`.